processor/transform/tables-to-jsx.ts

```diff
@@ -0,0 +1,98 @@
+import { Parents, Table, TableCell, Text } from 'mdast';
+import { visit, EXIT } from 'unist-util-visit';
+import { Transform } from 'mdast-util-from-markdown';
+
+import { mdxJsx } from 'micromark-extension-mdx-jsx';
+import { fromMarkdown } from 'mdast-util-from-markdown';
+import { mdxJsxFromMarkdown } from 'mdast-util-mdx-jsx';
+import { phrasing } from 'mdast-util-phrasing';
+
+const visitor = (table: Table, index: number, parent: Parents) => {
+  let hasFlowContent = false;
+
+  const tableCellVisitor = (cell: TableCell) => {
+    if (!phrasing(cell.children[0])) {
+      hasFlowContent = true;
+      return EXIT;
+    }
+
+    visit(cell, 'text', (text: Text) => {
+      if (text.value.match(/\n/)) {
+        hasFlowContent = true;
+        return EXIT;
+      }
+    });
+  };
+
+  visit(table, 'tableCell', tableCellVisitor);
+  if (!hasFlowContent) return;
+
+  const head = {
+    type: 'mdxJsxFlowElement',
+    name: 'thead',
+    children: [
+      {
+        type: 'mdxJsxFlowElement',
+        name: 'tr',
+        children: table.children[0].children.map((cell, idx) => {
```

Comment:
This assumes the first row will always be a `<thead>`, right? And that's the same as how it works today, yeah?

Reply:
Yea, the markdown tables seem to have a very specific format.